src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -12,8 +12,8 @@ public static class Enumerable
     {
         public static IEnumerable<TSource> Where<TSource>(this IEnumerable<TSource> source, Func<TSource, bool> predicate)
         {
-            if (source == null) throw Error.ArgumentNull("source");
-            if (predicate == null) throw Error.ArgumentNull("predicate");
+            Error.CheckSourceNotNull(source);
+            Error.CheckPredicateNotNull(predicate);
```

Comment:
I'm not sure this change to hide the check-and-throw in a method is an improvement.  I agree it helps clean up the code, but in the past I believe I've seen this have implications on inlining, so while this could be a small readability improvement, I think it's possible it could also be a small performance hit.  I could be remembering incorrectly, or maybe it's so small in this context that it doesn't matter, but it makes me hesitate.

Regardless, I'd prefer to see such a change separated out into its own PR so that it can be more easily evaluated on its own.